### PR TITLE
properly handle caching of usernames from uids

### PR DESCRIPTION
**Pull request by ericfranz**
Fixes https://github.com/OSC/ondemand/issues/1119

### CODE REVIEW

File: apps/dashboard/app/models/files.rb [hunk 1]
```diff
@@ -144,14 +144,21 @@ def self.mime_type_by_extension(path)
   end
 
   def self.username(uid)
-    @username_for_ids ||= Hash.new do |h, key|
-      h[key] = begin
-        Etc.getpwuid(uid).name
```
Comment (johrstrom) [inline apps/dashboard/app/models/files.rb:149]: Actually ya know what, this memorization patter does work. Only this needs to be `Etc.getpwuid(key).name` (using the key) not the field `uid`.

File: apps/dashboard/app/models/files.rb [hunk 2]
```diff
@@ -144,14 +144,21 @@ def self.mime_type_by_extension(path)
...
-      rescue
-        uid
```
Comment (johrstrom) [inline apps/dashboard/app/models/files.rb:151]: similarly here, we need to use `key` not `uid`.

**Comment by ericfranz** [comment]
Do you find:

```ruby
  def self.username_from_cache(uid)
    @username_for_ids ||= Hash.new do |h, key|
      h[key] = username(key)
    end
    
    @username_for_ids[uid]
  end
```

more readable than

```ruby
  def self.username_from_cache(uid)
    @username_for_ids ||= {}

    if @username_for_ids.key?(uid)
      @username_for_ids[uid]
    else
      @username_for_ids[uid] = username(uid)
    end
  end
```

**Comment by johrstrom** [comment]
I think the former does need so Ruby experience for sure to be able to tell what's going on. But I think you need to program for the language that you're in right? The former seems more Ruby idiomatic, right?

**Pull request merged**